internal/services/databricks/databricks_virtual_network_peering_resource.go

```diff
@@ -0,0 +1,356 @@
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Azure ARM databricks virtual network peering creation.")
+	var id vnetpeering.VirtualNetworkPeeringId
+
+	// I need to include the workspace ID in the properties because I need the name
+	// of the workspace to create the peerings ID
+	workspaceId, err := vnetpeering.ParseWorkspaceID(d.Get("workspace_id").(string))
+	if err != nil {
+		return fmt.Errorf("unable to parse 'workspace_id': %+v", err)
+	}
+
+	id = vnetpeering.NewVirtualNetworkPeeringID(subscriptionId, d.Get("resource_group_name").(string), workspaceId.WorkspaceName, d.Get("name").(string))
+
+	existing, err := client.Get(ctx, id)
+	if err != nil {
+		if !response.WasNotFound(existing.HttpResponse) {
+			return fmt.Errorf("checking for presence of existing Databricks %s: %s", id, err)
+		}
+	}
+
+	if !response.WasNotFound(existing.HttpResponse) {
+		return tf.ImportAsExistsError("azurerm_databricks_virtual_network_peering", id.ID())
+	}
+
+	allowForwardedTraffic := d.Get("allow_forwarded_traffic").(bool)
+	allowGatewayTransit := d.Get("allow_gateway_transit").(bool)
+	allowVirtualNetworkAccess := d.Get("allow_virtual_network_access").(bool)
+	useRemoteGateways := d.Get("use_remote_gateways").(bool)
+	remoteVirtualNetwork := d.Get("remote_virtual_network_id").(string)
+	databricksAddressSpace := utils.ExpandStringSlice(d.Get("address_space_prefixes").([]interface{}))
+	remoteAddressSpace := utils.ExpandStringSlice(d.Get("remote_address_space_prefixes").([]interface{}))
+
+	props := vnetpeering.VirtualNetworkPeeringPropertiesFormat{
+		DatabricksAddressSpace: &vnetpeering.AddressSpace{
+			AddressPrefixes: databricksAddressSpace,
+		},
+		// The RP always creates the same vNet ID for the Databricks internal vNet in the below format:
+		// '/subscriptions/{subscription}/resourceGroups/{group1}/providers/Microsoft.Network/virtualNetworks/workers-vnet'
+		DatabricksVirtualNetwork: &vnetpeering.VirtualNetworkPeeringPropertiesFormatDatabricksVirtualNetwork{
+			Id: utils.String(networkParse.NewVirtualNetworkID(id.SubscriptionId, id.ResourceGroupName, "workers-vnet").ID()),
+		},
+		RemoteAddressSpace: &vnetpeering.AddressSpace{
+			AddressPrefixes: remoteAddressSpace,
+		},
+		RemoteVirtualNetwork: vnetpeering.VirtualNetworkPeeringPropertiesFormatRemoteVirtualNetwork{
+			Id: utils.String(remoteVirtualNetwork),
+		},
+		AllowForwardedTraffic:     &allowForwardedTraffic,
+		AllowGatewayTransit:       &allowGatewayTransit,
+		AllowVirtualNetworkAccess: &allowVirtualNetworkAccess,
+		UseRemoteGateways:         &useRemoteGateways,
+	}
+
+	peer := vnetpeering.VirtualNetworkPeering{
+		Name:       &id.VirtualNetworkPeeringName,
+		Properties: props,
+	}
+
+	locks.ByID(databricksVnetPeeringsResourceType)
+	defer locks.UnlockByID(databricksVnetPeeringsResourceType)
```

Comment:
presumably the lock wants moving up to the top?

Also, per ARM we don't need to set Name btw, the API should be ignoring/rejecting it per the ARM Spec:

```suggestion
	locks.ByID(databricksVnetPeeringsResourceType)
	defer locks.UnlockByID(databricksVnetPeeringsResourceType)
	
	existing, err := client.Get(ctx, id)
	if err != nil {
		if !response.WasNotFound(existing.HttpResponse) {
			return fmt.Errorf("checking for presence of existing Databricks %s: %s", id, err)
		}
	}

	if !response.WasNotFound(existing.HttpResponse) {
		return tf.ImportAsExistsError("azurerm_databricks_virtual_network_peering", id.ID())
	}

	allowForwardedTraffic := d.Get("allow_forwarded_traffic").(bool)
	allowGatewayTransit := d.Get("allow_gateway_transit").(bool)
	allowVirtualNetworkAccess := d.Get("allow_virtual_network_access").(bool)
	useRemoteGateways := d.Get("use_remote_gateways").(bool)
	remoteVirtualNetwork := d.Get("remote_virtual_network_id").(string)
	databricksAddressSpace := utils.ExpandStringSlice(d.Get("address_space_prefixes").([]interface{}))
	remoteAddressSpace := utils.ExpandStringSlice(d.Get("remote_address_space_prefixes").([]interface{}))

	props := vnetpeering.VirtualNetworkPeeringPropertiesFormat{
		DatabricksAddressSpace: &vnetpeering.AddressSpace{
			AddressPrefixes: databricksAddressSpace,
		},
		// The RP always creates the same vNet ID for the Databricks internal vNet in the below format:
		// '/subscriptions/{subscription}/resourceGroups/{group1}/providers/Microsoft.Network/virtualNetworks/workers-vnet'
		DatabricksVirtualNetwork: &vnetpeering.VirtualNetworkPeeringPropertiesFormatDatabricksVirtualNetwork{
			Id: utils.String(networkParse.NewVirtualNetworkID(id.SubscriptionId, id.ResourceGroupName, "workers-vnet").ID()),
		},
		RemoteAddressSpace: &vnetpeering.AddressSpace{
			AddressPrefixes: remoteAddressSpace,
		},
		RemoteVirtualNetwork: vnetpeering.VirtualNetworkPeeringPropertiesFormatRemoteVirtualNetwork{
			Id: utils.String(remoteVirtualNetwork),
		},
		AllowForwardedTraffic:     &allowForwardedTraffic,
		AllowGatewayTransit:       &allowGatewayTransit,
		AllowVirtualNetworkAccess: &allowVirtualNetworkAccess,
		UseRemoteGateways:         &useRemoteGateways,
	}

	peer := vnetpeering.VirtualNetworkPeering{
		Properties: props,
	}
```

Reply:
Fixed.